### PR TITLE
Set Citadel as the default provider for Pilot certificate

### PR DESCRIPTION
**Pull request by lei-tang**
Please provide a description for what this PR is for: https://github.com/istio/istio/issues/19950.

As some platforms may not have k8s signing APIs, set Citadel as the default provider for Pilot certificate.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -129,10 +129,7 @@ var (
 		"number of attributes for stackdriver")
 	stackdriverTracingMaxNumberOfMessageEvents = env.RegisterIntVar("STACKDRIVER_TRACING_MAX_NUMBER_OF_MESSAGE_EVENTS", 200, "Sets the "+
 		"max number of message events for stackdriver")
-	// TODO (lei-tang): the default value of this option is currently set as "kubernetes" to be consistent
-	// with the existing istiod implementation and testing. As some platforms may not have k8s signing APIs,
-	// we may change the default value of this option as "citadel".
-	pilotCertProvider = env.RegisterStringVar("PILOT_CERT_PROVIDER", "kubernetes",
+	pilotCertProvider = env.RegisterStringVar("PILOT_CERT_PROVIDER", "",
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/main.go]: s/""/"citadel"

File: pilot/pkg/features/pilot.go [hunk 2]
```diff
@@ -286,10 +286,7 @@ var (
 	IstiodService = env.RegisterStringVar("ISTIOD_ADDR", "",
 		"Service name of istiod. If empty the istiod listener, certs will be disabled.")
 
-	// TODO (lei-tang): the default value of this option is currently set as "kubernetes" to be consistent
-	// with the existing istiod implementation and testing. As some platforms may not have k8s signing APIs,
-	// we may change the default value of this option as "citadel".
-	PilotCertProvider = env.RegisterStringVar("PILOT_CERT_PROVIDER", "kubernetes",
+	PilotCertProvider = env.RegisterStringVar("PILOT_CERT_PROVIDER", "",
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go]: ditto

File: pilot/cmd/pilot-agent/main.go (hunk 1 above)
Comment (lei-tang) [reply pilot/cmd/pilot-agent/main.go]: Done.

File: pilot/pkg/features/pilot.go (hunk 2 above)
Comment (lei-tang) [reply pilot/pkg/features/pilot.go]: Done.

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
run `BUILD_WITH_CONTAINER=1 make gen`. It wasn't needed before since we hadn't fixed the tests yet

File: manifests/gateways/istio-ingress/templates/deployment.yaml [hunk 3]
```diff
@@ -284,6 +286,10 @@ spec:
           - name: SDS_ENABLED
             value: "{{ .Values.global.sds.enabled }}"
           volumeMounts:
+{{- if eq .Values.global.pilotCertProvider "citadel" }}
+          - mountPath: /etc/istio/citadel-ca-cert
```
Comment (myidpt) [inline manifests/gateways/istio-ingress/templates/deployment.yaml]: Can you explain why we want to mount the cert? Isn't Pilot using the cert generated by Citadel running inside it?

Comment (lei-tang) [reply manifests/gateways/istio-ingress/templates/deployment.yaml]: Pilot uses the cert generated by Citadel. The mouting of the cert is for the TLS between data plane and control plane.

**Comment by howardjohn** [comment]
what if there is no citadel? can pilot create this?

**Comment by lei-tang** [comment]
The Citadel referred here is the Citadel linked in Pilot. Even without the legacy Citadel, Pilot can obtained a certificate issued from the Citadel linked in Pilot.

**Comment by howardjohn** [comment]
oh I read the pr wrong, I thought pilot was mounting it but that's ingress. Sounds good, ignore me

**Pull request merged**